src/shared/hooks/useNSQuery.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { getParamsFromURL } from '../../uniprotkb/utils/resultsUtils';
+import { getAPIQueryUrl } from '../config/apiUrls';
+import { Column } from '../config/columns';
+
+import useNS from './useNS';
+
+const useNSQuery = ({
+  queryColumns,
+  facets,
```

Comment:
Couldn't we get those 2 from inside the hook?

Reply:
facets are derived from the location, right? And queryColumns from the namespace and the userPreference

Reply:
- `queryColumns` also depend on the `ViewMode` so I wanted to wait until later to see if it can be moved.
- `facets`, I seem to have somehow completely lost in the refactor, having a look!

Reply:
>     * `queryColumns` also depend on the `ViewMode` so I wanted to wait until later to see if it can be moved.
Yep, that's what I meant when I said userPreference, so you could call that hook in here too and that should give that information to you.

Reply:
Ah yes of course! I weirdly had assumed it was in the state, not the user preferences